src/Xamarin.Android.Build.Tasks/Tasks/GetAotArguments.cs

```diff
@@ -320,9 +320,13 @@ string GetLdFlags (NdkTools ndk, AndroidTargetArch arch, int level, string toolP
 
 					string libstubsPath = Path.Combine (AndroidBinUtilsDirectory, relPath, "libstubs", ArchToRid (arch)).Replace (" ", "\\ ");
 
+					ldFlags.Append ("\\\"");
 					ldFlags.Append (Path.Combine (libstubsPath, "libc.so"));
-					ldFlags.Append (' ');
+					ldFlags.Append ("\\\"");
+					ldFlags.Append (" \\\"");
```

Comment:
Would this code (like the ones in the section above https://github.com/xamarin/xamarin-android/blob/63db768f7cf8925f623a0228f2705a30a6d0783a/src/Xamarin.Android.Build.Tasks/Tasks/GetAotArguments.cs#L293  be better?

```
				libs.Add (Path.Combine (androidLibPath, "libc.so"));
				libs.Add (Path.Combine (androidLibPath, "libm.so"));

				ldFlags.Append ($"\\\"{string.Join ("\\\";\\\"", libs)}\\\"");
```

It looks like we might be able to remove some duplicate code there...

Reply:
I tried this, and the AOT still failed, tripping on spaces :(

Reply:
@grendello I was testing this today and have a diff that appears to be working similar to what @dellis1972 suggested (at least on Windows). I also updated a couple of LLVM/NDK conditions to try to make it a bit easier to follow:

 * https://gist.github.com/pjcollins/067fc494251c7271e4d8d74dfe4afbd3


The only "Failed to load shared library..." [message I am seeing now](https://gist.github.com/pjcollins/93f02c784a9266a0cc36d56f3c4e0cc5#file-gistfile1-txt-L216) mentions `libLLVM.so`, which may still of concern?

Reply:
Thanks @pjcollins, updated the PR, let's see how it works :)

Reply:
I'm also seeing this message on d17-3 classic so this can probably be ignored, at least with respect to the changes in this PR:

> 08-05 18:35:03.581 10257 10257 I monodroid-assembly: Failed to load shared library 'libLLVM.so'. dlopen failed: library "libLLVM.so" not found